docs/DeveloperGuide.md

```diff
@@ -1,34 +1,357 @@
+* [6. Appendix: Requirements](#6-appendix-requirements)
+    + [6.1 Product scope](#61-product-scope)
+    + [6.2 User stories](#62-user-stories)
+    + [6.3 Use cases](#63-use-cases)
+    + [6.4 Non-Functional Requirements](#64-non-functional-requirements)
+    + [6.5 Glossary](#65-glossary)
+* [7. Appendix: Instructions for manual testing](#7-appendix-instructions-for-manual-testing)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## 1. Introduction
 
-## Product scope
-### Target user profile
+
+
+### 1.1. Purpose
+
+This document specifies the architecture and software design decisions taken to develop our application, CheatLogs. 
+Targeted towards developers who are or want to work on CheatLogs.
+
+### 1.2. Product Scope
+
+This document will cover the high-level architecture of the program, as well as the details for the implementation of the features.
+
+#### 1.2.1. Target User Profile
 
 {Describe the target user profile}
 
-### Value proposition
+#### 1.2.2. Value Proposition
 
 {Describe the value proposition: what problem does it solve?}
 
-## User Stories
+#### 1.3. Design Goals
+
+Our main goal is to design a cheatsheet management system that will help novice and beginner programmers to learn and adapt to various programming languages. 
+Our cheatsheet app comes with a set of preloaded cheatsheets. 
+Additionally, the user can input their own cheat sheets into the application. 
+This will make cheatlogs a comprehensive, one-stop solution for every student to manage their cheatsheet.
+
+#### 1.4. Definitions
+
+
+
+## 2. Setting Up, Getting Started
+
+
+
+### 2.1. Prerequisites
+
+1. JDK 11.0.8
+2. Java IDE (IntelliJ Recommended)
+
+CheatLogs has been developed on Java 11 and may not be supported on other versions. 
+You are free to use any java file editor and run the program by following the step by step guide below. 
+
+### 2.2. Running the Project
+
+1. Ensure you have Java 11 or above installed in your Computer.
+2. Download the latest cheatlogs.jar from here.
+3. Move the file to a folder you want to use as the home folder for this application.
+4. Invoke java -jar cheatlogs.jar on the command line to run the program. A welcome message should appear, as shown below.
+
+![](Images/Image1.PNG)
+
+### 2.3 Importing into IntelliJ [optional]
+
+In case you prefer to use IDEs to develop. 
+IntelliJ IDEA community edition is a popular free choice. 
+After installing a version of it you can import it using the “Get from Version Control” option below and clone from our [repository](https://github.com/AY2021S1-CS2113T-W11-3/tp.).
+
+![](Images/Image2.PNG)
+
+## 3. Design
+
+This section will elaborate on the architecture and component design of CheatLogs.
+
+### 3.1. Architecture
+
+The image below illustrates the high-level design of CheatLogs.
+
+![](Images/Image3.PNG)
```

Comment:
Maybe consider increasing the font of the text in this Architecture diagram? This could improve the readability of the diagram